osu.Game.Rulesets.Mania/Mods/ManiaKeyMod.cs

```diff
@@ -5,31 +5,304 @@
 using System.Linq;
 using osu.Game.Beatmaps;
 using osu.Game.Rulesets.Mania.Beatmaps;
+using osu.Game.Rulesets.Mania.Objects;
 using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
 
 namespace osu.Game.Rulesets.Mania.Mods
 {
-    public abstract class ManiaKeyMod : Mod, IApplicableToBeatmapConverter
+    public abstract class ManiaKeyMod : Mod, IApplicableToBeatmapConverter, IApplicableAfterBeatmapConversion
     {
         public override string Acronym => Name;
         public abstract int KeyCount { get; }
         public override ModType Type => ModType.Conversion;
         public override double ScoreMultiplier => 1; // TODO: Implement the mania key mod score multiplier
         public override bool Ranked => UsesDefaultConfiguration;
 
+        private double shortestJack { get; set; }
+
         public void ApplyToBeatmapConverter(IBeatmapConverter beatmapConverter)
         {
             var mbc = (ManiaBeatmapConverter)beatmapConverter;
 
-            // Although this can work, for now let's not allow keymods for mania-specific beatmaps
             if (mbc.IsForCurrentRuleset)
                 return;
 
             mbc.TargetColumns = KeyCount;
         }
 
-        public override Type[] IncompatibleMods => new[]
+        public void ApplyToBeatmap(IBeatmap beatmap)
+        {
+            if (beatmap.BeatmapInfo.Ruleset.ShortName != "mania")
```

Comment:
I don't think this check is needed since there is no valid case this mod is ever used outside of mania.

Reply:
This mod convert std maps to different key counts.
And this conversion shouldn't apply for those